yarn-project/simulator/src/avm/journal/trace.ts

```diff
@@ -0,0 +1,175 @@
+import { Fr } from '@aztec/foundation/fields';
+
+export class WorldStateAccessTrace {
+  public accessCounter: number;
+  //public contractCalls: Array<TracedContractCall> = [];
```

Comment:
What are the commented things place holder for ?

Reply:
Before I started breaking this all down into multiple PRs, I gave it a shot all-at-once [here](yarn-project/simulator/src/avm/journal/trace.ts).